### PR TITLE
SPARKC-706: Add basic support for Cassandra vectors

### PR DESCRIPTION
**Pull request by jacek-lewandowski**
# Description

## How did the Spark Cassandra Connector Work or Not Work Before this Patch

Describe the problem, or state of the project that this patch fixes. Explain
why this is a problem if this isn't obvious.

Example: 
  "When I read from tables with 3 INTS I get a ThreeIntException(). This is a problem because I often want to read from a table with three integers."

## General Design of the patch

How the fix is accomplished, were new parameters or classes added? Why did you
pursue this particular fix?

Example: "I removed the incorrect assertion which would throw the ThreeIntException. This exception was incorrectly added and the assertion is not actually needed."

Fixes: [Put JIRA Reference HERE](https://datastax-oss.atlassian.net/projects/SPARKC)

# How Has This Been Tested?

Almost all changes and especially bug fixes will require a test to be added to either the integration or Unit Tests. Any tests added will be automatically run on travis when the pull request is pushed to github. Be sure to run suites locally as well.

# Checklist:

- [ ] I have a ticket in the [OSS JIRA](https://datastax-oss.atlassian.net/projects/SPARKC)
- [ ] I have performed a self-review of my own code
- [ ] Locally all tests pass (make sure tests fail without your patch)

### CODE REVIEW

File: driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala [hunk 1]
```diff
@@ -768,6 +769,29 @@ object TypeConverter {
...
+    val elemConverter = implicitly[TypeConverter[T]]
+
+    implicit def elemTypeTag: TypeTag[T] = elemConverter.targetTypeTag
+
+    @transient
+    lazy val targetTypeTag = {
+      implicitly[TypeTag[CqlVector[T]]]
+    }
+
+    def newCollection(items: Iterable[Any]): java.util.List[T] = {
```
Comment (jtgrabowski) [inline driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala:782]: Could be private

File: driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala [hunk 2]
```diff
@@ -768,6 +769,29 @@ object TypeConverter {
...
+      val buf = new java.util.ArrayList[T](dimension)
```
Comment (jtgrabowski) [inline driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala:783]: Why java collection here?

**Review by jtgrabowski — COMMENTED** [review]
Great work! :)

Please have a look at the minor comments and add a note with a large font in the main README stating that the vector type is now supported. Additionally, please add a document showcasing how it can be used (examples taken from tests should be okay here).

File: connector/src/it/scala/com/datastax/spark/connector/rdd/typeTests/VectorTypeTest.scala [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  override def cassandraOnly(f: => Unit): Unit = super.cassandraOnly(f)
+
+  override lazy val conn = CassandraConnector(sparkConf)
+
+  val VectorTable = "vectors"
+
+  def createVectorTable(session: CqlSession, table: String): Unit = {
+    session.execute(
+      s"""CREATE TABLE IF NOT EXISTS $ks.$table (
+         |  id INT PRIMARY KEY,
```
[the remark this answers is not in the document]
Comment (jacek-lewandowski) [reply connector/src/it/scala/com/datastax/spark/connector/rdd/typeTests/VectorTypeTest.scala:39]: Perhaps, why do you ask?

File: driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala (hunk 2 above)
Comment (jacek-lewandowski) [reply driver/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala:783]: This collection is passed to `CqlVector.getInstance` which expects either Java `List` or array. If we use here a Scala collection and then let `JavaConverters` adjust that, we will have to create one extra wrapper which is a waste in this case.

File: connector/src/it/scala/com/datastax/spark/connector/rdd/typeTests/VectorTypeTest.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (jtgrabowski) [reply connector/src/it/scala/com/datastax/spark/connector/rdd/typeTests/VectorTypeTest.scala:31]: Wondering if that's supported.

File: doc/2_loading.md [hunk 4]
```diff
@@ -184,7 +184,49 @@ val street = address.getString("street")
 val number = address.getInt("number")
 ```
 
-[//]: # (TODO loading vectors)
+### Reading vectors
```
Comment (jtgrabowski) [inline doc/2_loading.md:187]: Should we add similar docs to https://github.com/datastax/spark-cassandra-connector/blob/master/doc/14_data_frames.md ?

**Review by jtgrabowski — APPROVED** [review]
LGTM!

**Pull request merged**